Bottom: ba7213fb35a5861f507bf19ed8c49798cb782ed2
Top:    01d1104959349777ab0427b95b4ad28d573f63bf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-04-01 13:35:50 +0100

Fix leak of cross vertex connectivity lists when deleting contours

Free these lists as we free the contour's nodes in poly_DelContour()


---

diff --git a/src/polygon1.c b/src/polygon1.c
index c0dd8e7..cc00888 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1901,12 +1901,25 @@ poly_ClrContour (PLINE * c)
 void
 poly_DelContour (PLINE ** c)
 {
-  VNODE *cur;
+  VNODE *cur, *prev;
 
   if (*c == NULL)
     return;
-  for (cur = (*c)->head.prev; cur != &(*c)->head; free (cur->next))
-    cur = cur->prev;
+  for (cur = (*c)->head.prev; cur != &(*c)->head; cur = prev)
+    {
+      prev = cur->prev;
+      if (cur->cvc_next != NULL)
+        {
+          free (cur->cvc_next);
+          free (cur->cvc_prev);
+        }
+      free (cur);
+    }
+  if ((*c)->head.cvc_next != NULL)
+    {
+      free ((*c)->head.cvc_next);
+      free ((*c)->head.cvc_prev);
+    }
   /* FIXME -- strict aliasing violation.  */
   if ((*c)->tree)
     {
